src/config/navigation/tabConfig.tsx

```diff
@@ -102,3 +106,31 @@ export const tabNavigatorConfig: TabNavigatorConfig = {
     aboutTabConfig
   ]
 };
+
+export const createDynamicTabConfig = (
+  accessibilityLabel: string,
+  iconName: string,
+  iconSize: number,
+  index: number,
+  label: string,
+  length: number,
+  screen: ScreenName,
+  initialParams?: Record<string, any>
+): TabConfig => ({
+  stackConfig: defaultStackConfig({
+    initialParams,
+    initialRouteName: screen,
+    isDrawer: false
+  }),
+  tabOptions: {
+    tabBarAccessibilityLabel: `${accessibilityLabel || label} (Tab ${index + 1} von ${length})`,
+    tabBarLabel: label,
+    tabBarIcon: ({ color }: TabBarIconProps) => (
+      <OrientationAwareIcon
+        color={color}
+        Icon={Icon[upperFirst(iconName)]}
+        size={normalize(iconSize)}
+      />
```

Comment:
what if an icon would need more specific props like

```
<OrientationAwareIcon
  color={color}
  Icon={Icon.About}
  landscapeStyle={{ marginRight: -normalize(6) }}
  size={normalize(28)}
  style={{ marginTop: normalize(3) }}
/>
```

Reply:
I didn't add a customization option because we don't always need these props, but it can be added. I will develop this feature.

Reply:
True, it is not needed always, but sometimes :D